src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx

```diff
@@ -0,0 +1,176 @@
+  // user info
+  const balance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceByAssetId(state, { assetId: vaultTokenId }),
+  )
+
+  const cryptoAmountAvailable = bnOrZero(balance).div(`1e${asset.precision}`)
+  const fiatAmountAvailable = bnOrZero(cryptoAmountAvailable).times(marketData.price)
+
+  const selectedLocale = useAppSelector(selectSelectedLocale)
+  const descriptionQuery = useGetAssetDescriptionQuery({ assetId, selectedLocale })
+
+  const walletAddress = useWalletAddress('0x0000000000000000000000000000000000000000')
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!(vaultAddress && api)) return
+        const [opportunity] = await Promise.all([
+          api.findByOpportunityId(
+            toAssetId({ chainId, assetNamespace, assetReference: vaultAddress }),
+          ),
+        ])
+        if (!opportunity) {
+          return toast({
+            position: 'top-right',
+            description: translate('common.somethingWentWrongBody'),
+            title: translate('common.somethingWentWrong'),
+            status: 'error',
+          })
+        }
+        setOpportunity(opportunity)
+
+        const claimableTokens = await opportunity.getClaimableTokens(walletAddress)
+        setClaimableTokens(claimableTokens)
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('IdleOverview error:', error)
+      }
+    })()
+  }, [api, vaultAddress, chainId, toast, translate, walletAddress])
+
+  const additionalParams: Record<string, any> = {}
+
+  useAppSelector(selectorState => {
+    if (claimableTokens && claimableTokens.length > 0) {
+      let rewardAssets: any[] = []
+      claimableTokens.forEach(token => {
+        const rewardAsset = selectAssetById(selectorState, token.assetId)
+        if (rewardAsset) {
+          rewardAssets.push({
+            ...rewardAsset,
+            cryptoBalance: bnOrZero(token.amount).div(`1e+${asset.precision}`).toPrecision(),
+          })
+        }
+      })
+
+      additionalParams.rewardAssets = rewardAssets
+    }
+  })
+
+  const totalClaimableRewards = useMemo(() => {
+    if (!claimableTokens) return bnOrZero(0)
+    return claimableTokens.reduce((totalRewards, token) => {
+      totalRewards = totalRewards.plus(token.amount)
+      return totalRewards
+    }, bnOrZero(0))
+  }, [claimableTokens])
+
+  const claimDisabled = !totalClaimableRewards || totalClaimableRewards.lte(0)
+
+  if (!opportunity) {
+    return (
+      <Center minW='500px' minH='350px'>
+        <CircularProgress />
+      </Center>
+    )
+  }
+
+  const menu: DefiButtonProps[] = [
+    {
+      label: 'common.deposit',
+      icon: <ArrowUpIcon />,
+      action: DefiAction.Deposit,
+    },
+    {
+      label: 'common.withdraw',
+      icon: <ArrowDownIcon />,
+      action: DefiAction.Withdraw,
+    },
+  ]
+
+  if (!opportunity.metadata.cdoAddress) {
+    menu.push({
```

Comment:
This is still wrong.
Same as you can't have `let` variables within a component that you mutate, you can't mutate the intrinsic value of (array) objects within a component. It is defined as a `const` and since objects are accessed by reference, and react uses `===` comparison, it doesn't matter whether or not you push to it, its reference will never change.
This might rerender the way you expect, but not for the reason you'd expect. The reason why it might "work" would be because another dependency made the reconciliation algorithm decide it should rerender this component.

`useMemo()` / `useState()` are your friends for stable references and invalidating these conditionally.

Reply:
Looking back into this, the `opportunity` shouldn't change so we shouldn't even need to be programmatic at all across renders here?